backend/api/pdf_progress.py

```
@@ -0,0 +1,294 @@
+    
+    def disconnect(self, session_id: str):
+        """Remove a WebSocket connection"""
+        if session_id in self.active_connections:
+            del self.active_connections[session_id]
+        if session_id in self.progress_data:
+            del self.progress_data[session_id]
+        # Also remove from Redis
+        if self.use_redis:
+            try:
+                self.redis.delete(f"pdf_progress:{session_id}")
+            except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)